.github/workflows/regression-tests.yml

```diff
@@ -230,14 +241,22 @@ jobs:
         sudo make install
 
       # Windows Testing
-    - name: Install cocotb build dependencies (Windows)
-      if: startsWith(matrix.os, 'windows')
-      run: conda install --yes -c msys2 m2-base m2-make m2w64-toolchain libpython
-    - name: Install cocotb (Windows)
-      if: startsWith(matrix.os, 'windows')
-      run: python -m pip install  --global-option build_ext --global-option --compiler=mingw32 -e .
     - name: Install Python testing dependencies (Windows)
       run: pip install coverage xunitparser pytest pytest-cov
+    - name: Install cocotb build dependencies (Windows, mingw)
+      if: startsWith(matrix.os, 'windows') && matrix.toolchain == 'mingw'
+      run: conda install --yes -c msys2 m2-base m2-make m2w64-toolchain libpython
+    - name: Install cocotb (Windows, mingw)
+      if: startsWith(matrix.os, 'windows') && matrix.toolchain == 'mingw'
+      run: python -m pip install  --global-option build_ext --global-option --compiler=mingw32 -e .
+      continue-on-error: ${{matrix.may_fail || false}}
+    - name: Install cocotb (Windows, msvc)
+      if: startsWith(matrix.os, 'windows') && matrix.toolchain == 'msvc'
+      run: python -m pip install -e .
+      continue-on-error: ${{matrix.may_fail || false}}
+    - name: Install cocotb runtime dependencies (Windows, msvc)
+      if: startsWith(matrix.os, 'windows') && matrix.toolchain == 'msvc'
+      run: conda install --yes -c msys2 m2-base m2-make
```

Comment:
Only msvc was allowed here to fail, mingw not, as `may_fail` is not set in the matrix. I did a few more tweaks in #1798 and I decided to continue development there. For msvc I moved the installation of msys after the compilation as I think it should compile without it mingw/msys. When something like #1562 lands, this could be of advantage.